sonic-chassisd/scripts/chassisd

```diff
@@ -746,26 +1226,84 @@ class ChassisdDaemon(daemon_base.DaemonBase):
         else:
             self.log_warning("Caught unhandled signal '{}' - ignoring...".format(SIGNALS_TO_NAMES_DICT[sig]))
 
+    def submit_dpu_callback(self, module_index, admin_state):
+        try_get(self.module_updater.chassis.get_module(module_index).set_admin_state, admin_state, default=False)
+        pass
+
+    def set_initial_dpu_admin_state(self):
+        """Send admin_state trigger once to modules those are powered up"""
+        threads = []
+        for module_index in range(0, self.module_updater.num_modules):
+            op = None
+            # Get midplane state of DPU
+            module_info_dict = self.module_updater._get_module_info(module_index)
+            module_name = self.platform_chassis.get_module(module_index).get_name()
+            state_db = daemon_base.db_connect("STATE_DB")
+            midplane_table = swsscommon.Table(state_db, CHASSIS_MIDPLANE_INFO_TABLE)
+            fvs = midplane_table.get(module_name)
+            midplane_state = 'False'
+            if isinstance(fvs, list) and fvs[0] is True:
+                fvs = dict(fvs[-1])
+                midplane_state = fvs[CHASSIS_MIDPLANE_INFO_ACCESS_FIELD]
```

Comment:
We are checking for the `midplane_state` here in the DB (and if it is not present in the DB we are assuming that the  module is not reachable since `midplane_state=False` by default) but in case of Nvidia platform the `midplane_state` is being updated to the DB by Chassisd itself, so chassisd here is checking for midplane state in DB before it is supposed to update the midplane state itself. Either we need to re order code so that we update midplane_state to db before we check the DB for values, or instead of checking the DB we can directly invoke the midplane reachability API

Reply:
@gpunathilell  Using dpu  operational_state and the API to get this as planned.